Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,779 @@
+                Write-Host "ERROR: Input file: [$($InputFilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+
+            Write-Host "Fetching all Azure Service Bus Namespace(s) from [$($InputFilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $inputServiceBusNampespaces = Import-Csv -LiteralPath $InputFilePath
+
+            $validServiceBusNampespaces = $inputServiceBusNampespaces | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+
+            $validServiceBusNampespaces | ForEach-Object {
+                $resourceId = $_.ResourceId
+
+                try {                
+                    $currentServiceBusNamespaceDetails = Get-AzServiceBusNamespace -ResourceGroupName $_.ResourceGroupName -NamespaceName $_.ResourceName -ErrorAction SilentlyContinue
+                    $currentTLSVersion = ($currentServiceBusNamespaceDetails).MinimumTlsVersion
+                    $serviceBusNamespacesDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $currentServiceBusNamespaceDetails.Id } },
+                    @{N = 'ResourceGroupName'; E = { $currentServiceBusNamespaceDetails.Id.Split("/")[4] } },
+                    @{N = 'ResourceName'; E = { $currentServiceBusNamespaceDetails.Name } }, 
+                    @{N = 'MinimumTLSVersion'; E = { $currentTLSVersion } }
+
+                }
+                catch {
+                    Write-Host "Error fetching Azure Service Bus Namespace details for resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+    }
+
+    $serviceBusNamespacesCount = ($serviceBusNamespacesDetails | Measure-Object).Count
+
+    if ($serviceBusNamespacesCount -eq 0) {
+        Write-Host "No Azure Service Bus Namespace(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$($serviceBusNamespacesCount)] Azure Service Bus Namespace(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    Write-Host $([Constants]::SingleDashLine)
+
+    # List for actionable service bus namespace(s)
+    $actionableServiceBusNamespacesDetails = @()
+
+    Write-Host "Filtering Azure Service Bus Namespace(s) for which secure TLS version is not secure..." -ForegroundColor $([Constants]::MessageType.Info)
+
+    $serviceBusNamespacesDetails | ForEach-Object {
+        if (-not (CheckIfCurrentTLSVersionIsSecure($_.MinimumTLSVersion))) {
+            $actionableServiceBusNamespacesDetails += $_
+        }
+    }
+
+    $actionableServiceBusNamespacesDetailsCount = ($actionableServiceBusNamespacesDetails | Measure-Object).Count
+
+    if ($actionableServiceBusNamespacesDetailsCount -eq 0) {
```

Comment:
Here we need to log other service bus namespaces as skipped right? If we end here, in log file, both remediated and skipped namespaces will be empty which would be wrong.

Reply:
Yup you are right, thanks!